src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -100,15 +109,36 @@ internal void Add(HeaderDescriptor descriptor, IEnumerable<string?> values)
         {
             ArgumentNullException.ThrowIfNull(values);
 
+            // It's relatively common to only add a single value with this overload, especially when copying
+            // between HttpHeaders collections. Avoid boxing the enumerator and possibly a HeaderStoreItemInfo
+            // allocation by deferring to the overload for a single value instead.
+            if (values is IList<string?> { Count: 1 } valuesList)
```

Comment:
Are the values guaranteed to be already grouped by the header name/descriptor when they get here? Or is it possible for this method to be called twice with different values of the same header?

Reply:
Thanks!